### ISSUE TITLE
bug: docs for plugin definition syntax missing attributes

### ISSUE THREAD

**Issue by pnadolny13** [meltano/meltano#6588]
I was reading https://docs.meltano.com/reference/plugin-definition-syntax and noticed that there are some attributes that are missing from the syntax list. We might not be wanting to list every possible option but a few are missing that are important, comparing against the hub schemas https://github.com/meltano/hub/tree/main/schemas/plugin_definitions:

- executable
- settings_group_validation
- prereq
- description
- definition
- etc. There might be more but thats what I saw from a quick skim

**Comment by tayloramurphy** [comment]
@pnadolny13 good catch. @edgarrmondragon @aaronsteers is there a way to automate validation of these docs with teh JSON schema  at some point in the future so that these don't drift?

**Comment by edgarrmondragon** [comment]
> is there a way to automate validation of these docs with teh JSON schema at some point in the future so that these don't drift?

@tayloramurphy There are no tools that I know of to do this easily that play well with Jekyll, but we could always just use templating + walking the JSON schema tree to generate the page.

Though, the schema for Hub plugin definitions lives in `meltano/hub` and it's split by plugin type so that makes it harder to use here in `meltano/meltano`. Also, my personal preference, since we usually edit those schema files by hand, would be for them to be written in yaml instead of json, since it's slightly more readable.

**Comment by tayloramurphy** [comment]
Still relevant though it looks like `prereq` is the only one missing. 

Note you may want to update this when adding gold / silver / bronze @pnadolny13

**Comment by tayloramurphy** [comment]
Still relevant

**Comment by edgarrmondragon** [comment]
Actually addressed by:

- #9758
